### PR TITLE
Sanitize JSON output to remove nulls

### PR DESCRIPTION
**Pull request by jcantrill**
[test]

cc @fbricon @adietish

### CODE REVIEW

File: src/main/java/com/openshift/internal/util/JBossDmrExtentions.java [hunk 1]
```diff
@@ -29,6 +34,59 @@
...
+		if(node.getType() == ModelType.OBJECT) {
+			Collection<String> emptyKeys = new ArrayList<>(node.keys().size());
+			for (String key : node.keys()) {
+				ModelNode child = node.get(key);
+				if(child.getType() == ModelType.UNDEFINED) {
+					node.remove(key);
+				}else {
+					sanitize(child);
+				}
+				if(child.getType() == ModelType.OBJECT && child.keys().size() == 0) {
```
Comment (fbricon) [inline src/main/java/com/openshift/internal/util/JBossDmrExtentions.java]: do you ever reach that statement?

**Comment by fbricon** [comment]
LTGM but Travis seems to disagree.

My only concern is if we sanitize genuine and significant null values from the server. Better be sure it doesn't bite us in the ass

**Comment by jcantrill** [comment]
[test]

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**